MdePkg/Include/Pi/PiHob.h

```diff
@@ -267,6 +267,7 @@ typedef UINT32 EFI_RESOURCE_ATTRIBUTE_TYPE;
 #define EFI_RESOURCE_ATTRIBUTE_WRITE_PROTECTED      0x00000100
 #define EFI_RESOURCE_ATTRIBUTE_EXECUTION_PROTECTED  0x00000200
 #define EFI_RESOURCE_ATTRIBUTE_PERSISTENT           0x00800000
+#define EFI_RESOURCE_ATTRIBUTE_SPECIAL_PURPOSE      0x08000000

```

Comment:
This change was added in edk2 here - https://github.com/tianocore/edk2/commit/b538d6a1b1d79d0dc2fd68da9e19b5e257d3dc5a.

Can you please cherry-pick that into your branch prior to the changes in Gcd.c?